### PR TITLE
Added some documentation on testing

### PR DESCRIPTION
**Pull request by hackaugusto**
This adds some documentation regarding the testing system. It is just a first introduction, I may expand it later.

### CODE REVIEW

File: docs/testing.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
+How is Raiden tested.
+#####################
+
+Raiden tests are split into these categories. These categories are ordered from
```
Comment (LefterisJP) [inline docs/testing.rst]: nitpick: these -> the following

File: docs/testing.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+the faster and easier to debug to the slower and harder to debug:
+
+- Unit tests: A collection of fast tests that do not require external services
+  to run. Because these tests don't require any services nor threads to run,
+  they don't have problems with flakyness and are easy to debug. These tests
+  are located under `raiden/tests/unit`.
+- Property tests: These tests use randomly generated events. These tests are
```
Comment (LefterisJP) [inline docs/testing.rst]: Nitcpik: `These tests are -> They are`

File: docs/testing.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  important ensure the system does not break when an unexpected series of
+  events happen. Because of the architecture used by the project these tests
+  are deterministic and do not require threads, therefore they are not flaky,
+  but can be harder to debug because of their random nature. These tests are
+  located under `raiden/tests/fuzz`.
+- Smoketest: This is a quick self contained test, that can run by user or
```
Comment (LefterisJP) [inline docs/testing.rst]: `can run by user` -> `can be ran by a user`

File: docs/testing.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  nodes, and the REST API for these nodes. A fresh set of smart contracts are
+  deployed into this private chain, including Tokens and Raiden's smart
+  contracts. These tests are important to make sure Raiden will work as
+  expected with the Ethereum clients, however because they orchestrate multiple
+  systems they are considerably slower than the previous categories,
+  susceptible to flakyness, and harder to debug. These tests are located under
+  `raiden/tests/integration`.
+- Scenario tests: These tests run multiple times against the available
+  testnets. They exercise the Raiden nodes through the available REST API, in
+  the same a user application would. These tests are important to test the
```
Comment (LefterisJP) [inline docs/testing.rst]: `the same a user` -> `the same way a user`

File: docs/testing.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  system against a real life conditions (using the internet). These tests use
```
Comment (LefterisJP) [inline docs/testing.rst]: `against a real life conditions` -> `against real life conditions`

File: docs/testing.rst [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  the scenario-player utility, and the scenarios are available under
+  `raiden/tests/scenarios`.
+
+Unit tests
```
Comment (LefterisJP) [inline docs/testing.rst]: Why not put the above bullet points into those sections from the beginning? And have one section for each test type?

File: docs/testing.rst [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+==========
+
+Unit tests are deterministic, single threaded, and side-effect free. Because
+these tests don't interact with other systems there isn't any orchestration nor
+synchornization, so the tests cannot be flaky and are easy to debug.
```
Comment (LefterisJP) [inline docs/testing.rst]: `synchornization` -> `synchronization`

File: docs/testing.rst [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+This the preferred type of test for business logic, and should always be the
+preferred type of test to add to Raiden's test suite.
+
+Integration Tests
+=================
+
+These tests exist to make sure that Raiden works properly with the other
+systems. Including the Ethereum client, Matrix, and the serivices ofr path
```
Comment (LefterisJP) [inline docs/testing.rst]: `serivices ofr` -> `services for`

File: docs/testing.rst [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+finding and monitoring channels.
+
+These tests are written in python and rely heavily on the pytest fixture
+system. This is necessary because these tests will run multiple Raiden nodes in
+a single process, exchange messages, and assert on the expected state of each
+individual node.
+
+Here is an overview of the fixture system:
```
Comment (LefterisJP) [inline docs/testing.rst]: Overview of fixture system could be its own subscection

File: docs/testing.rst [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+- Configuration variables are defined in `raiden/tests/fixtures/variables`.
+  This file contains the default settings for a test, these settings can be
```
Comment (LefterisJP) [inline docs/testing.rst]: `test, these` -> `test. These`

File: docs/testing.rst [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+  overwritten by the `pytest.mark.paremetrize` decorator to change the test
```
Comment (LefterisJP) [inline docs/testing.rst]: I think for RST you need double ``

File: docs/testing.rst [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  which configure how many channels each of these nodes should have at the
+  start of the test.
+- The fixtures responsible to start a private chain are defined in
+  `raiden/tests/integration/fixtures/blockchain.py`. These will start a new
+  private chain, either `geth` or `parity`, with `blockchain_number_of_nodes`
+  nodes. This private chain will have one prefunded account for each Raiden
+  node, which allows these nodes to send on-chain transactions.
+- The fixtures responsible to deploy the smart contracts are defined in
+  `raiden/tests/integration/fixtures/smart_contracts.py`. These fixtures depend
+  on the private chain, and will deploy the necessary Raidne and Token Smart
```
Comment (LefterisJP) [inline docs/testing.rst]: `Raidne` -> `Raiden`

File: docs/testing.rst [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  Contracts for the test.
+- The fixtures responsible to start the transport are defined in
+  `raiden/tests/integration/fixtures/transprot.py`. These fixtures will
```
Comment (LefterisJP) [inline docs/testing.rst]: `transprot` -> `transport`. Perhaps switch to links?

File: docs/testing.rst [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+  configure a local cluster of Matrix servers to run the tests.
+- The fixtures responsible to start the raiden nodes and open the initial
+  channels are defined in `raiden/tests/integration/fixtures/raiden_network.py`
+  . These fixtures depend on both the previous fixtures, the private
+  blockchain, the smart contract, and the transport fixtures. The raiden
+  network fixtures will then create the Raiden apps and open the necessary
+  channels for the test.
+
+Which of the above fixtures are used depends on the test function, and it is
+determined by pytest through it's fixture system. A test that uses a
```
Comment (LefterisJP) [inline docs/testing.rst]: `it's` -> `its`

File: docs/testing.rst [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+  contracts. These tests are important to make sure Raiden will work as
+  expected with the Ethereum clients, however because they orchestrate multiple
+  systems they are considerably slower than the previous categories,
+  susceptible to flakyness, and harder to debug. These tests are located under
+  `raiden/tests/integration`.
+- Scenario tests: These tests run multiple times against the available
+  testnets. They exercise the Raiden nodes through the available REST API, in
+  the same a user application would. These tests are important to test the
+  system against a real life conditions (using the internet). These tests use
+  the scenario-player utility, and the scenarios are available under
```
Comment (palango) [inline docs/testing.rst]: A link here would be nice.

**Pull request merged**